### ISSUE TITLE
Elevators do not work correctly when SpongeForge is installed on the Server - Also rarely occurs without sponge.

### ISSUE THREAD

**Issue by D3nnis3n** [Thutmose/Thut#71]
I just tested this too (Sponge was **_NOT_** enabled when i got, reported and tested the EcologyMod Desync Bug and the Advanced Rocketry Bug!), and found out your mod does not work right when SpongeForge is installed. It seems to bug around intstead of moving or just moves to the lowest Point it can and get's stuck there. I couldn't extract much out of the spam of Errors in the log. So you might try it oyurself. Afaik most mods should be able to be compatible with it nowadays, even the car mod is :D

**Comment by Thutmose** [comment]
can you include a paste of the spam of errors?

**Comment by D3nnis3n** [comment]
I try to do so after i tested the new orespawn Version through.

**Comment by D3nnis3n** [comment]
The spam does not get logges and is so fast that it is really hard to copy it out of console

**Comment by D3nnis3n** [comment]
Okay, the spam was not related to Sponge. But yeh, the issue persists and NO Messages are logged at all. It behaves nearly ecactly like the Advanced Rocketry issue. The Elevator will jsut get stuck - but you won't wall through the Floor. And it does work correct occasionally. I make you another Video.

**Comment by D3nnis3n** [comment]
http://minecraft.godsofwar.de/video3.mp4

**Comment by Thutmose** [comment]
Ok, and this only occurs when sponge is installed?

**Comment by D3nnis3n** [comment]
Yes. If i remove sponge everything works fine for at least the two minutes i tested driving up, down and all around.

**Comment by D3nnis3n** [comment]
Also since the last Version if i reskin the Controller it is visible immediatly and not only after reconnect - thanks for that fix that somehow got in oO

**Comment by Thutmose** [comment]
In your sponge settings, try disabling all of the entity related settings and see if that does anything.

I did find why the reskin wasn't synced immediately and fixed that.

**Comment by D3nnis3n** [comment]
Like that?

>     entity {
>         # Number of colliding entities in one spot before logging a warning. Set to 0 to disable
>         collision-warn-size=0
>         # Number of entities in one dimension before logging a warning. Set to 0 to disable
>         count-warn-size=0
>         # Number of ticks before a painting is respawned on clients when their art is changed
>         entity-painting-respawn-delay=0
>         # Number of ticks before the fake player entry of a human is removed from the tab list (range of 0 to 100 ticks).
>         human-player-list-remove-delay=0
>         # Controls the time in ticks for when an item despawns.
>         item-despawn-rate=0
>         # The upper bounded range where living entities farther from a player will likely despawn
>         living-hard-despawn-range=0
>         # The amount of seconds before a living entity between the soft and hard despawn ranges from a player to be considered for despawning
>         living-soft-despawn-minimum-life=0
>         # The lower bounded range where living entities near a player may potentially despawn
>         living-soft-despawn-range=0
>         # Max size of an entity's bounding box before removing it. Set to 0 to disable
>         max-bounding-box-size=0
>         # Square of the max speed of an entity before removing it. Set to 0 to disable
>         max-speed=0

**Comment by D3nnis3n** [comment]
Nope, did not help.
Also, may just be random. But i only found this Happening on the highest Floor. It did not yet occur on any other Floors. And as you can see it starts working again after some time. Maybe that helps somehow.

**Comment by Thutmose** [comment]
Ya those, also check for other optimizations that might be affecting entity updates, it maybe datamanager things?

**Comment by D3nnis3n** [comment]
I'll do so. I also linked your issue to one on their github. Maybe they can help. I hope so.

**Comment by Thutmose** [comment]
sponge is somehow making it desync:

![image](https://user-images.githubusercontent.com/2363472/31843010-d6e3a0ba-b5be-11e7-9478-1e7b016192cf.png)

I told it to go down, and the top light turned off, and a few seconds later the bottom light turned on, so the server has it down at the bottom, the client has not been notified of this yet.

Relogging results in this:
![image](https://user-images.githubusercontent.com/2363472/31843043-fe7a1d52-b5be-11e7-80f2-bb7604c32af5.png)

So somehow the elevator's position or movement isn't being synced to the client.  This should be synced via a combination of vanilla entity syncing, as well as some datamanager values used by the elevator itself (for coordinating which floor is being gone to for the elevator's button lighting, which does also seem to be a bit buggy when the elevator glitches on the client like that, you can see it by looking at the colours of the buttons on the elevator, it still thinks the elevator is trying to move on the client, when they server says it is at destination, on relog the buttons have correct lighting)

**Comment by Thutmose** [comment]
The reason why collision is working fine, is that on SMP, the player's movement wrt collision is almost entirely handled on the client itself, this is why flyhacks work, so the player is colliding off the client side version of the elevator, so they walk on it just fine.  The earlier bug where you would fall through it was an error in the collision code itself related to AR's gravity stuff and the elevator not properly accounting for some cases where its velocity was set.

**Comment by D3nnis3n** [comment]
Thanks for your great explanations, very interesting. As i linked the issue over there i just hope they'll see them, or would you like to post them there, as this seems like something being altered by sponge?

**Comment by Thutmose** [comment]
I will try to look into it some more on my end before I take it to them, as I would like to try to narrow down what seem to be breaking first.

**Comment by D3nnis3n** [comment]
Alright, over at sponge it was suggested to look if it is in spawn protection or a plugin is activated that interferes and cancels Events. I got None of both.

**Comment by Thutmose** [comment]
It seems this issue isn't specific to sponge, I had it occur once on my testing server, so it can occur without sponge, just very rarely

**Comment by D3nnis3n** [comment]
I could not yet reproduce it without sponge. There at least is a coincidence it happens way more often with sponge, then.

**Comment by Thutmose** [comment]
It has only happened once so far on my non-sponge server, over a few hundred uses of the elevator...

**Comment by phit** [comment]
I wasn't able to reproduce this with Sponge 1.12.2-2529-7.0.0-BETA-2732 and thuttech-6.0.7

**Comment by Thutmose** [comment]
Hmm, has anything entity sync related changed in sponge lately?  I had this bug happen once without sponge, out of a whole lot of elevator use, so it isn't specific to sponge, most likely a vanilla or forge sync issue that was somehow worse with sponge

**Comment by phit** [comment]
Nah, nothing was changed to my knowledge. OP may have had some weird config options enabled that are recommend to be disabled with mods, like tile entity activation ranges. Without knowing their `global.conf`, Sponge versions etc. this is pretty much impossible to debug.

**Comment by D3nnis3n** [comment]
I had the latest Version of sponge to that date. I update all my mods every day. Config in respect to tile Things can be seen above. Dunno which Information is missing. Did not Report this issue to sponge, but instead here and Thut was able to reproduce himself, so ... yeh. If you Need something from me, just ask me.

**Comment by phit** [comment]
your sponge global.conf, specifically you wanna disable entity activation range for thuttech

http://vpaste.net/nlrOo

**Comment by D3nnis3n** [comment]
https://gist.github.com/D3nnis3n/a958a45d757f75a74bfabede3c7c5066

Here you go.

**Comment by phit** [comment]
yeah, no idea your settings seem fine, try setting the values to 0 instead of just enabled=false maybe that setting is borked

**Comment by D3nnis3n** [comment]
That didn't work out. After several tries the same issue occurs rarely again. Funny bug.
